### ISSUE TITLE
Pagination links

### ISSUE THREAD

**Issue by zamith** [jeregrine/jsonapi#104]
As far as I could tell the QueryParser will correctly parse pagination data, but there is not way to then add pagination links (`next`, `prev`, `first` and/or `last`) to the links object at the collection level, as per the JSON:API spec.

Am I missing something or is it really a feature that is missing? If so, would you be willing to take a PR on that?

**Comment by zamith** [comment]
Created a PR for this #107.

**Comment by zamith** [comment]
Even though the PR is merged, there is still one outstanding issue with it. I'll paste what I wrote on the PR:

> The issue here is that I need access to the pagination data on the links overridable function, so I can do something like this:

```elixir
def links(data, conn, %{cursor: cursor}) do
  %{
    next: url_for_pagination(data, conn, %{cursor: cursor})
  }
end
```

> Here it's being passed in the meta, if I try to pass it in the data like so:

```elixir
render(conn, "index.json", %{data: %{collection: collection, pagination: pagination}})
```

> Then the collection will be passed in on its entirety to the view, instead of being iterated over. That is because of this:

```elixir
def encode_data(view, data, conn, query_includes) when is_list(data) do
  Enum.map_reduce(data, [], fn d, acc ->
    {to_include, encoded_data} = encode_data(view, d, conn, query_includes)
    {to_include, acc ++ [encoded_data]}
  end)
end
```

> which will not be called, since data is not a list.

> Does that make sense, or am I missing something?

**Comment by zamith** [comment]
I have the code ready for this, that I've been running, but wanted to get your take on it first. Please let me know how to proceed.

**Comment by doomspork** [comment]
@zamith how about opening a PR?  We can always close the PR if it's decided not to merge.

**Comment by jherdman** [comment]
I believe this should be fully resolved now via #164 . I'm going to close this issue, @zamith . Feel free to reopen if you find the issue still unresolved.

**Comment by lucacorti** [comment]
Is this supposed to work? If so, how do you add pagination links?

I've enabled `with_pagination` in the config, but the responses don't include any links and I get in the logs:

```
[info] You've set with_pagination but have not defined any pagination links, thus no pagination links will be returned
```

I'm passing pagination parameters to Scrivener which returns a page/size/total data which I should pass back to the view somehow, but didn't find any hint in the docs on how to do this. The `links` callback from `JSONAPI.View` only processes the data, so I guess pagination links shpould not be manually generated there.
